tuf/repository_lib.py

```diff
@@ -1442,10 +1456,32 @@ def generate_targets_metadata(targets_directory, target_files, version,
         shutil.copyfile(target_path, digest_target)
 
   # Generate the targets metadata object.
-  targets_metadata = tuf.formats.TargetsFile.make_metadata(version,
-      expiration_date, filedict, delegations)
-
-  return targets_metadata
+  # Use generalized build_dict_conforming_to_schema func to produce a dict that
+  # contains all the appropriate information for targets metadata,
+  # checking that the result conforms to the appropriate schema.
+  # TODO: Later, probably after the rewrite for TUF Issue #660, generalize
+  #       further, upward, by replacing generate_targets_metadata,
+  #       generate_root_metadata, etc. with one function that generates
+  #       metadata, possibly rolling that upwards into the calling function.
+  #       There are very few things that really need to be done differently.
+  if delegations is not None:
+    return tuf.formats.build_dict_conforming_to_schema(
+        tuf.formats.TARGETS_SCHEMA,
+        version=version,
+        expires=expiration_date,
+        targets=filedict,
+        delegations=delegations)
+  else:
+    return tuf.formats.build_dict_conforming_to_schema(
+        tuf.formats.TARGETS_SCHEMA,
+        version=version,
+        expires=expiration_date,
+        targets=filedict)
```

Comment:
So I couldn't sell the [`kwargs` alternative to you](https://github.com/awwad/tuf/pull/23#discussion_r269024916)? :)

Reply:
Whatcha mean?  The extras bit (e.g. delegations) in the calls below this one?

Reply:
The function uses kwargs and returns deepcopy(kwargs) (after some modifications).

Reply:
Gonna merge and use this in some other work, now that the rest has been addressed, but if I missed something, please LMK.  I'll keep the extras-style optional arguments construction in mind if there are other places calls like this start to need to happen.

Reply:
Oh yes, I was referring to passing something like an `extra_kwargs` that may or may not contain the `delegations` (sorry for being unclear).